stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/dao/MessageDao.kt

```diff
@@ -4,55 +4,90 @@ import androidx.room.Dao
+    @Transaction
+    abstract suspend fun messagesForChannelNewerThan(
+        cid: String,
+        limit: Int = 100,
+        dateFilter: Date,
+    ): List<MessageEntity>
 
     @Query("SELECT * from stream_chat_message WHERE cid = :cid AND createdAt >= :dateFilter ORDER BY createdAt ASC LIMIT :limit")
-    suspend fun messagesForChannelEqualOrNewerThan(cid: String, limit: Int = 100, dateFilter: Date): List<MessageEntity>
+    @Transaction
+    abstract suspend fun messagesForChannelEqualOrNewerThan(
+        cid: String,
+        limit: Int = 100,
+        dateFilter: Date,
+    ): List<MessageEntity>
 
     @Query("SELECT * from stream_chat_message WHERE cid = :cid AND createdAt < :dateFilter ORDER BY createdAt DESC LIMIT :limit")
-    suspend fun messagesForChannelOlderThan(cid: String, limit: Int = 100, dateFilter: Date): List<MessageEntity>
+    @Transaction
+    abstract suspend fun messagesForChannelOlderThan(
+        cid: String,
+        limit: Int = 100,
+        dateFilter: Date,
+    ): List<MessageEntity>
 
     @Query("SELECT * from stream_chat_message WHERE cid = :cid AND createdAt <= :dateFilter ORDER BY createdAt DESC LIMIT :limit")
-    suspend fun messagesForChannelEqualOrOlderThan(cid: String, limit: Int = 100, dateFilter: Date): List<MessageEntity>
+    @Transaction
+    abstract suspend fun messagesForChannelEqualOrOlderThan(
+        cid: String,
+        limit: Int = 100,
+        dateFilter: Date,
+    ): List<MessageEntity>
 
     @Query("SELECT * from stream_chat_message WHERE cid = :cid ORDER BY createdAt DESC LIMIT :limit")
-    suspend fun messagesForChannel(cid: String, limit: Int = 100): List<MessageEntity>
+    @Transaction
+    abstract suspend fun messagesForChannel(cid: String, limit: Int = 100): List<MessageEntity>
 
     @Query("DELETE from stream_chat_message WHERE cid = :cid AND createdAt < :deleteMessagesBefore")
-    suspend fun deleteChannelMessagesBefore(cid: String, deleteMessagesBefore: Date)
+    @Transaction
+    abstract suspend fun deleteChannelMessagesBefore(cid: String, deleteMessagesBefore: Date)
 
     @Query("DELETE from stream_chat_message WHERE cid = :cid AND id = :messageId")
-    suspend fun deleteMessage(cid: String, messageId: String)
-
-    @Query(
-        "SELECT * FROM stream_chat_message " +
-            "WHERE stream_chat_message.id IN (:ids)"
-    )
-    suspend fun select(ids: List<String>): List<MessageEntity>
-
-    @Query(
-        "SELECT * FROM stream_chat_message " +
-            "WHERE stream_chat_message.id IN (:id)"
-    )
-    suspend fun select(id: String?): MessageEntity?
-
-    @Query(
-        "SELECT * FROM stream_chat_message " +
-            "WHERE stream_chat_message.syncStatus IN (:syncStatus) ORDER BY createdAt ASC"
-    )
-    suspend fun selectSyncNeeded(syncStatus: SyncStatus = SyncStatus.SYNC_NEEDED): List<MessageEntity>
+    @Transaction
```

Comment:
From the `Transaction` class docs:
"Putting this annotation on an {@link Insert}, {@link Update} or {@link Delete} method has no impact because those methods are always run inside a transaction. "

Reply:
Yes, you're right. Nice catch 👍